Update repository docs for swift-version support and recent CI check changes (#2815)
apple/swift-nio: CONTRIBUTING.md

### Make sure your patch works for all supported versions of swift

> The CI will do this for you.  Currently all versions of swift >= 5.8 are supported.

Reviewer: ```suggestion
The CI will do this for you.  Currently all versions of Swift >= 5.8 are supported.
```

Not new in your code, but Swift is a proper noun.

> The CI will do this for you.  Currently all versions of Swift >= 5.8 are supported.

Reviewer 2: It's probably worth adding a note that the CI in the first PR requires a project maintainer to kick it off.

> The CI will do this for you, but a project maintainer must kick it off for you.  Currently all versions of Swift >= 5.8 are supported.

Reviewer 3: I wonder if we should just have a single place where supported versions are defined and refer to that.  Or specify as last 3 minor versions here like it is elsewhere to stop us having to update multiple places every 6 months.

Author: I wondered that too, but it felt like a more significant rejig to perhaps do in a more structured way next?

> If you wish to test this locally you have two options [act](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://github.com/nektos/act) and docker compose files.

Reviewer: ```suggestion
If you wish to test this locally you have two options [act](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://github.com/nektos/act) and Docker Compose files.
```

> If you wish to test this locally you have two options [act](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://github.com/nektos/act) and Docker Compose files.

Reviewer 4: Why is this link redirecting through Google?

```suggestion
If you wish to test this locally you have two options [act](https://github.com/nektos/act) and Docker Compose files.
```

Author: oops, copy link fail!

#### Act

> [Install act](https://nektosact.com) and then you can run the full suite of checks via:

Reviewer 5: A nit, but do we want to link to the installation page? Would require less fumbling around the website.
```suggestion
[Install act](https://nektosact.com/installation/) and then you can run the full suite of checks via:
```

```
act pull_request
```

> Note that NIOSwift matrix testing makes use of nightly builds, so you may want to make use of the ```--action-offline-mode``` to avoid repulling those.

Reviewer: ```suggestion
Note that SwiftNIO matrix testing makes use of nightly builds, so you may want to make use of the ```--action-offline-mode``` to avoid repulling those.
```

> #### Doccker compose files

Reviewer: ```suggestion
#### Docker Compose files
```

> You can use the docker-compose files.    For example usage of docker compose see the main [README](./README.md#an-alternative-using-docker-compose)

Reviewer: ```suggestion
You can use the docker-compose files.  For example usage of Docker Compose see the main [README](./README.md#an-alternative-using-docker-compose)
```

Try to keep your lines less than 120 characters long so github can correctly display your changes.

> SwiftNIO now uses the [swift-format](https://github.com/swiftlang/swift-format) tool to bring consistency to code formatting.    Their is a specific [.swift-format](./.swift-format) configuration file.    This will be checked and enforced on PRs, note that the check will run on the current most recent stable version target which may not match that in your own local development environment.

Reviewer: ```suggestion
SwiftNIO uses the [swift-format](https://github.com/swiftlang/swift-format) tool to bring consistency to code formatting.  There is a specific [.swift-format](./.swift-format) configuration file.  This will be checked and enforced on PRs.  Note that the check will run on the current most recent stable version target which may not match that in your own local development environment.
```

> If you want to apply the formatting to your local repo before commit then you can either run [check-swift-format.sh](./scripts/check-swift-format.sh) which will use your current tool chain, or to match the CI checks exactly you can use act (see above):

Reviewer: ```suggestion
If you want to apply the formatting to your local repo before you commit then you can either run [check-swift-format.sh](./scripts/check-swift-format.sh) which will use your current toolchain, or to match the CI checks exactly you can use `act` (see above):
```

I recommend turning `(see above)` into an interior link.
